yarn-project/ethereum/src/test/fallback_transport.test.ts

```diff
@@ -117,4 +116,42 @@ describe('fallback_transport', () => {
       fail('Transaction should have a block hash');
     }
   });
+
+  it('does not fall back when encountering a contract error', async () => {
+    // Deploy a simple contract that will revert
+    // This is a minimal contract that always reverts with "ALWAYS_REVERT"
+    const bytecode = '0x6080604052348015600f57600080fd5b50603f80601d6000396000f3fe6080604052600080fdfe';
+
+    // Deploy the contract
+    const deployHash = await client.sendTransaction({
+      data: bytecode,
+    });
+    const deployReceipt = await client.waitForTransactionReceipt({ hash: deployHash });
+    expect(deployReceipt.status).toBe('success');
+    const contractAddress = deployReceipt.contractAddress;
+    expect(contractAddress).toBeDefined();
+
+    // this should revert
+    try {
+      await client.sendTransaction({
+        to: contractAddress,
+        data: '0xcafe',
+      });
+
+      fail('Transaction should have reverted');
+    } catch (error: any) {
+      // The error should be from the first node
+      expect(error.message).toContain('revert');
+
+      // Wait a sec to ensure any potential transaction would have been processed
+      await sleep(1000);
+
+      // Check that no transactions were sent to the contract on the second node
+      const block1 = await publicClient1.getBlock();
+      const block2 = await publicClient2.getBlock();
+
+      expect(block1.transactions.length).toBe(1);
+      expect(block2.transactions.length).toBe(0);
+    }
```

Comment:
How does this work? We're sending the tx but not awaiting its receipt. Is it possible the revert we're getting back corresponds to a simulation (or gas estimation), instead of the actual tx? And the `block1.transactions.length` corresponds to the contract deployment?